tests/unit/apps/test_auxiliary.py

```diff
@@ -1171,12 +1188,19 @@ async def test_ceph_osd_verify_nova_compute_fail(mock_lookup, model):
         charm="ceph-osd",
         channel="octopus/stable",
         config={"source": {"value": "distro"}},
-        machines={},
+        machines=machines,
         model=model,
         origin="ch",
         series="focal",
         subordinate_to=[],
-        units={},
+        units={
+            f"ceph-osd/{i}": Unit(
+                name=f"ceph-osd/{i}",
+                workload_version="17.0.1",
+                machine=machines[f"{i}"],
+            )
+            for i in range(3)
+        },
```

Comment:
[non-blocking] I don't understand how these parts relate to this PR

Reply:
Long story short: the changes does not direct affect this, but I was playing around and identified this issue

Long story:
I was making some changes on `is_valid_track` for auxiliary charms and in some point I started checking if the current track was inside the possible_tracks ``and some unit tests started failing because on theory you can't have ceph-osd with workload version on `pacific` (N)  and the channel on `octopus` (N-1), so I've changed the workload_version to be properly on octopus.

After some though I realize that we can't be strict like this because on a transition (E.g: octopus to pacific) you can have a channel that does not match with the current workload_version. It would make sense to check that the channel is on N or N+1 in relation to the workload but never N-1 like it was here. This check is not easy to implement right now,